saleor/tax/migrations/0004_migrate_vatlayer.py

```diff
@@ -0,0 +1,187 @@
+    use_origin_country_map = {}
+
+    for vatlayer_config in vatlayer_configs:
+        config_dict = {
+            item["name"]: item["value"] for item in vatlayer_config.configuration
+        }
+        channel = vatlayer_config.channel
+
+        # Migrate `countries_to_calculate_taxes_from_origin` setting.
+        origin_country = _clear_country_code(config_dict["origin_country"])
+        countries_to_calculate_taxes_from_origin = _clear_str_list_country_codes(
+            config_dict["countries_to_calculate_taxes_from_origin"]
+        )
+        for country in countries_to_calculate_taxes_from_origin:
+            use_origin_country_map[country] = origin_country
+
+        # Migrate `excluded_countries` to new tax configuration.
+        excluded_countries = _clear_str_list_country_codes(
+            config_dict["excluded_countries"]
+        )
+        if excluded_countries:
+            tax_configuration = channel.tax_configuration
+            if not tax_configuration:
+                tax_configuration = TaxConfiguration.objects.create(channel=channel)
+
+            for country in excluded_countries:
+                TaxConfigurationPerCountry.objects.update_or_create(
+                    tax_configuration=tax_configuration,
+                    country=country,
+                    defaults={"charge_taxes": False},
+                )
+
+    return use_origin_country_map
+
+
+def create_tax_rates(apps, use_origin_country_map):
+    TaxClass = apps.get_model("tax", "TaxClass")
+    TaxClassCountryRate = apps.get_model("tax", "TaxClassCountryRate")
+
+    tax_classes = TaxClass.objects.all()
+    vat_rates = VAT.objects.all()
+
+    rates = {}
+    for tax_class in tax_classes:
+        for vat in vat_rates:
+            # Collect standard rates to create
+            standard_rate = TaxClassCountryRate(
+                tax_class=tax_class,
+                country=vat.country_code,
+                rate=vat.data["standard_rate"],
+            )
+            rates[(tax_class.id, vat.country_code)] = standard_rate
+
+            # Collect reduced rates to create
+            if tax_class.name in vat.data["reduced_rates"]:
+                reduced_rate = TaxClassCountryRate(
+                    tax_class=tax_class,
+                    country=vat.country_code,
+                    rate=vat.data["reduced_rates"][tax_class.name],
+                )
+                rates[(tax_class.id, vat.country_code)] = reduced_rate
+
+        # Swap rates for countries that should use origin country tax rate instead of
+        # own rates.
+        for country_code, origin in use_origin_country_map.items():
+            country_rate_obj = rates.get((tax_class.id, country_code))
+            origin_rate_obj = rates.get((tax_class.id, origin))
+            if country_rate_obj and origin_rate_obj:
+                country_rate_obj.rate = origin_rate_obj.rate
+                rates[(tax_class.id, country_code)] = country_rate_obj
```

Comment:
I don't think that you need to re-assign it here

Reply:
I'm reassigning to override the original tax rate of a country with the rate from the "origin" country.